cilium/cmd/monitor.go

```diff
@@ -43,11 +43,23 @@ programs attached to endpoints and devices. This includes:
 	},
 }
 
+func listEventTypes() []string {
+	types := []string{}
+	for k := range eventTypes {
+		types = append(types, k)
+	}
+	return types
+}
+
 func init() {
 	RootCmd.AddCommand(monitorCmd)
 	monitorCmd.Flags().IntVarP(&eventConfig.NumCpus, "num-cpus", "c", runtime.NumCPU(), "Number of CPUs")
 	monitorCmd.Flags().IntVarP(&eventConfig.NumPages, "num-pages", "n", 64, "Number of pages for ring buffer")
 	monitorCmd.Flags().BoolVarP(&dissect, "dissect", "d", false, "Dissect packet data")
+	monitorCmd.Flags().StringVarP(&eventType, "type", "t", "", fmt.Sprintf("Filter by event types %v", listEventTypes()))
+	monitorCmd.Flags().Uint16Var(&fromSource, "from", 0, "filter by source endpoint id")
```

Comment:
To match the existing message format. Using other types would generate compile
time errors. The source fields from the message types define the
`.Source` format as `uint16`. 

See
https://github.com/cilium/cilium/blob/master/pkg/bpfdebug/drop.go#L30
https://github.com/cilium/cilium/blob/master/pkg/bpfdebug/debug.go#L129
https://github.com/cilium/cilium/blob/master/pkg/bpfdebug/debug.go#L229

also for `DstID` or `Arg1` (`uint32`) see

https://github.com/cilium/cilium/blob/master/pkg/bpfdebug/drop.go#L36
https://github.com/cilium/cilium/blob/master/pkg/bpfdebug/debug.go#L131
https://github.com/cilium/cilium/blob/master/pkg/bpfdebug/debug.go#L233

The ctypes are in 

https://github.com/cilium/cilium/blob/master/bpf/lib/common.h
https://github.com/cilium/cilium/blob/master/bpf/lib/dbg.h

but @borkmann and @tgraf can probably give a much better explanation on why
those fields are uint16 and uint32.